### PR TITLE
LPS-159024 [Raylife AP] Add logic for Dashboard Sales Chart calculations

### PR DESCRIPTION
**Pull request by joanagrego**
Related with --> [LPS-146090](https://issues.liferay.com/browse/LPS-146090)

### CODE REVIEW

**Comment by joanagrego** [comment]
ci:test:sf

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/dashboard-sales/index.css [hunk 1]
```diff
@@ -30,6 +30,18 @@
 	width: 19px;
 }
 
+.donut-chart-space-legend {
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/dashboard-sales/index.css:33]: Please change it here and change it everywhere else that makes use of it.
```suggestion
.dashboard-sales-space-legend {
```

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/dashboard-sales/index.js [hunk 2]
```diff
@@ -98,31 +307,56 @@ export default function () {
...
 
-	const getDateUntilGoal = getData()[0]?.dateUntilGoal;
-	const getSalesValue = getData()[0]?.salesValue;
-	const getGoalValue = getData()[0]?.goalValue;
-	const getSalesPercentual = getData()[0]?.salesPercentual;
-
 	const LegendElement = () => (
-		<div className="d-flex donut-chart-legend flex-column h-100 justify-content-end ml-5 mt-5">
+		<div className="d-flex donut-chart-space-legend flex-column h-100 justify-content-end mt-5">
 			<div className="donut-chart-screen font-weight-bolder h5">
```
Comment (silasbsi) [inline modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/dashboard-sales/index.js:333]: This class is specific to the donut chart component. Please check as if someone changes this class their code will be impacted

**Review by silasbsi — CHANGES_REQUESTED** [review]
LGTM :1st_place_medal: But, I added some suggestions.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/services/SalesGoal.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+	currentYear,
+	sixMonthsAgoDate,
+	threeMonthsAgoDate,
+} from '../utils/dateFormatter';
+import {axios} from './liferay/api';
+
+const DeliveryAPI = 'o/c/raylifesalesgoals';
+const userId = Liferay.ThemeDisplay.getUserId();
+
+export function getSalesGoalCurrentMonth() {
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/services/SalesGoal.js:26]: Done

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/dashboard-sales/index.css (hunk 1 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/dashboard-sales/index.css:33]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/dashboard-sales/index.js (hunk 2 above)
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/raylife-ap-react-fragments/fragments/dashboard-sales/index.js:333]: Done.

File: modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/services/Policy.js [hunk 4]
```diff
@@ -77,3 +62,27 @@ export function getPoliciesUntilCurrentMonthLastYear() {
 		`${DeliveryAPI}/?filter=policyStatus ne 'declined' and (startDate le ${oneYearAgoDate} and startDate ge ${lastYear}-01-01)&pageSize=200`
 	);
 }
+
+export function getPoliciesForSalesGoalCurrentMonth() {
```
[the remark this answers is not in the document]
Comment (joanagrego) [reply modules/apps/site-initializer/site-initializer-raylife-ap/src/main/resources/site-initializer/fragments/group/src/common/services/Policy.js:66]: Done.

**Comment by silasbsi** [comment]
LGTM :1st_place_medal: Thanks @joanagrego

**Review by silasbsi — APPROVED** [review]

**Review by eliasantosf — APPROVED** [review]
LGTM :fire: Thanks @joanagrego

**Comment by silasbsi** [comment]
ci:forward

**Comment by joanagrego** [comment]
ci:stop

**Comment by joanagrego** [comment]
ci:forward

**Pull request closed without merging**